embassy-net/src/udp.rs

```diff
@@ -143,8 +139,29 @@ impl<'a> UdpSocket<'a> {
     }
 }
 
-impl Drop for UdpSocket<'_> {
+#[cfg(feature = "igmp")]
+impl<'a, D: Driver + smoltcp::phy::Device + 'static> UdpSocket<'a, D> {
+    pub fn join_multicast_group<T>(&self, addr: T) -> Result<bool, smoltcp::iface::MulticastError>
```

Comment:
I don't think these methods should be in UdpSocket. Them being here implies the "joining" is scoped to the socket, ie if you drop the socket you un-join the group. This is not the case though.

these should be in `Stack` instead, mirroring smoltcp.

Reply:
I made the previous helpers on Stack public and reverted the changes on the Socket.